### PR TITLE
File upload

### PR DESCRIPTION
**Pull request by sei-byoung**
Allow files that have been uploaded to a view to be used as images for maps

### CODE REVIEW

**Review by sei-tspencer — APPROVED** [review]

File: src/app/components/map/map.component.ts [hunk 1]
```diff
@@ -83,18 +84,29 @@ export class MapComponent implements OnInit, OnChanges {
...
     this.vmMapsQuery.getById(this.mapIdInput).subscribe((data) => {
       this.name = data.name;
       this.mapId = data.id;
-      this.imageURL = data.imageUrl;
       this.teamIDs = data.teamIds;
 
+      // As in display component, need to figure out if the URL is b64 encoded and, if so, convert it to a blob
+      // and then generate an object url
+      const url = data.imageUrl;
+      console.log('URL = ' + url);
```
Comment (sei-ebram) [inline src/app/components/map/map.component.ts:96]: lots of console.logs here, how much do we care about clean consoles?

**Comment by sei-ebram** [comment]
In general there's lots of player-api generated code here.  @sei-aschlackman might need to weigh in on if there should have been a new player-api nuget package released that might transform this PR

File: src/app/components/map/map.component.ts (hunk 1 above)
Comment (sei-aschlackman) [reply src/app/components/map/map.component.ts:96]: Yea I would prefer to remove the log statements once dev is finished I think.

File: src/app/components/map/map-team-display/map-team-display.component.ts [hunk 2]
```diff
@@ -33,8 +34,24 @@ export class MapTeamDisplayComponent implements OnInit {
...
+    
+    this.vmMapsQuery.getById(this.mapId).subscribe(m => {
+      const url = m.imageUrl;
+      // If this is a base64 string, this image was uploaded to the view. Get a blob from the b64 string
+      // and use that to get an object url that will point to the image
+      
+      // Simple test for whether the string is an actual url. If not, it is b64 encoded.
+      if (!this.isURL(url)) {
+        const asBlob = this.b64ToBlob(url);
+        console.log('B64 as blob in display component:');
```
Comment (sei-aschlackman) [inline src/app/components/map/map-team-display/map-team-display.component.ts:46]: these log statements can probably be removed.

**Comment by sei-aschlackman** [comment]
> In general there's lots of player-api generated code here. @sei-aschlackman might need to weigh in on if there should have been a new player-api nuget package released that might transform this PR

I think it's fine as is. We only had generated code for the vm api in this project, but Ben needed to make some calls to the player api now. We haven't (yet?) made re-usable npm packages for that stuff. Not sure if it's worth the hassle or not.

**Pull request merged**